cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/service/http/DelayedHttpServiceResponder.java

```diff
@@ -190,6 +190,7 @@ private void emitMetrics(int status) {
   @Override
   public void close() {
     closed = true;
+    taskExecutor.releaseCallResources();
```

Comment:
I think this will release resources prematurely if BodyProducer is being used since the `execute(boolean keepAlive)` method call `close()` in the finally block.